retis-events/src/events.rs

```diff
@@ -122,13 +122,11 @@ impl Event {
 
 impl EventFmt for Event {
     fn event_fmt(&self, f: &mut std::fmt::Formatter, format: DisplayFormat) -> std::fmt::Result {
-        // First format the first event line starting with the always-there
-        // {common} section, followed by the {kernel} or {user} one.
-        write!(
-            f,
-            "{}",
-            self.0.get(&SectionId::Common).unwrap().display(format)
-        )?;
+        // First format the first event line starting with the {common} section,
+        // followed by the {kernel} or {user} one.
+        if let Some(common) = self.0.get(&SectionId::Common) {
+            write!(f, "{}", common.display(format))?;
+        }
         if let Some(kernel) = self.0.get(&SectionId::Kernel) {
             write!(f, " {}", kernel.display(format))?;
```

Comment:
In practice that can't happen but because no event section is mandatory now the code need to take it into account. You're right, we need to handle that case and not have a leading space. This can be done with `DelimWriter`.

However there's another issue: if a metadata event is displayed, it won't have any of the first sections and will be intended which is not what we want (I forgot to fix that), eg,

```
  Retis version v1.4.0-768e78200421-dirty

132005966153185 (0) [swapper/0] 0 [tp] net:netif_rx ...
```

This might be possible to solve with a `DelimWriter` too, forcing the first sub-section to be printed to not be intended. But if that first section is `skb` that might be a little surprising. The root cause is event sections are quite generic, but not in the way we print them, we have to make some assumptions. I think that's fine though, printing some of the first sections in a nice way is more an optimization than something we are forced to do.

Note: for the above to work the stack section might need some special handling, or be moved to the end of the output.

Reply:
+1 to use `DelimWriter`

Reply:
With the common section staying mandatory, the above is no longer a concern.